### PR TITLE
autoxflate: fix autoxflate on replication

### PR DESCRIPTION
**Pull request by sebsura**
### Thank you for contributing to the Bareos Project!

**Backport of PR #1576 to bareos-22**

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

File: systemtests/tests/autoxflate/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2021-2023 Bareos GmbH & Co. KG
```
Comment (pstorz) [inline systemtests/tests/autoxflate/CMakeLists.txt:3]:
```suggestion
#   Copyright (C) 2023-2023 Bareos GmbH & Co. KG
```

File: core/src/stored/device_resource.cc [hunk 2]
```diff
@@ -335,12 +335,23 @@ static bool ValidateGenericDevice(const DeviceResource& resource)
 
 bool DeviceResource::Validate()
 {
+  if (autodeflate != AutoXflateMode::IO_DIRECTION_NONE
+      && autodeflate_algorithm == 0) {
+    Jmsg(nullptr, M_ERROR, 0,
+         _("Device %s: If 'AutoDeflate' is set, then 'AutoDeflateAlgorithm' "
+           "also has to be set.\n"),
```
Comment (pstorz) [inline core/src/stored/device_resource.cc:342]:
```suggestion
         _("Device %s: 'AutoDeflate' is set but required 'AutoDeflateAlgorithm' is missing.\n"),
```

**Review by pstorz — CHANGES_REQUESTED** [review]
Looks good, please see comments

File: core/src/stored/device_resource.cc (hunk 2 above)
Comment (sebsura) [reply core/src/stored/device_resource.cc:342]: changed

**Review by pstorz — APPROVED** [review]

**Pull request merged**